contracts/ConsensusUtils.sol

```diff
@@ -114,37 +111,49 @@ contract ConsensusUtils is EternalStorage, ValidatorSet {
     }
   }
 
+  function _unBoundStake(address _staker) internal {
+    require(block.number > unboundingBlock(_staker));
+    _staker.transfer(unBoundingAmount(_staker));
+    _setUnBoundingAmount(_staker, 0);
+  }
+
   function _withdraw(address _staker, uint256 _amount, address _validator) internal {
     require(_validator != address(0));
     require(_amount > 0);
-    require(_amount <= stakeAmount(_validator));
-    require(_amount <= delegatedAmount(_staker, _validator));
+    if (unBoundingAmount(_staker) != 0)
```

Comment:
@Andrew-Pohl @leonprou 
this logic prevents multiple withdraw requests, and still breaks wrapping fuse staking for DEFI.
- user A requests to withdraw from the defi contract
- defi contract calls withdraw -> now unbounding amount of contract > 0
- user B requests to withdraw to withdraw BEFORE unbounding period is over
- defi contract calls withdraw, this IF resolves to true and _unboundStake is called and reverted because unbounding block has not reached yet

User B needs to wait for unbounding period before he can request a withdraw